File: Sources/OnScreen/Features/LeavesPoopStains.swift
import Foundation
import Yage

class LeavesPoopStains: LeavesTracesWhileWalking {
    override func install(on subject: Entity) {
        speciesForTraceEntities = .poopStain
        timeBetweenSpawns = 4
        traceExpirationTime = 17 / speciesForTraceEntities.fps
        super.install(on: subject)
    }
}

private extension Species {
    static let poopStain = Species(
        id: "poopstain",
        capabilities: [
            "AnimatedSprite",
            "AnimationsProvider",
            "PetsSpritesProvider"
        ],
        dragPath: "front",
        fps: 4,
        movementPath: "front",
        speed: 0,
        zIndex: -100
    )
}

File: Sources/windows/onscreen/features/leaves_poop_stains.py

from yage.models.species import Species
from .leaves_traces_while_walking import LeavesTracesWhileWalking


class LeavesPoopStains(LeavesTracesWhileWalking):
    def __init__(self, subject):
        super().__init__(subject)
        self.species_for_trace_entities = LeavesPoopStains.poop_stain_species()
        self.time_between_spawns = 4
        self.trace_expiration_time = 17 / self.species_for_trace_entities.fps

    @classmethod
    def poop_stain_species(cls) -> Species:
        return Species(
            id='poopstain',
            animations=[],
            capabilities=[
                "AnimatedSprite",
                "AnimationsProvider",
                "PetsSpritesProvider"
            ],
            drag_path='front',
            fps=4,
            movement_path='front',
            speed=0,
            tags=[],
            z_index=-100
        )
